File: python/section3/q22.py
# -*- coding: utf-8 -*-
from q20 import picup_eng_txt
import re


if __name__ == '__main__':
    txt = picup_eng_txt()
    pattern = r'\[\[Category:(\w+)\]\]'
    category_list = re.findall(pattern, txt)
    print(category_list)

File: scala/section3/src/main/scala/q22.scala
/**
  * Created by nomotoeriko on 2017/06/09.
  */
object q22 {
  def main(args: Array[String]): Unit = {
    val text = q20.load_data()
    val pattern = ".*\\[\\[Category:(.+)\\]\\].*".r
    pattern.findAllMatchIn(text).map(_.group(1)).foreach(println)
  }
}
